spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -437,6 +448,27 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 			};
 		}
 
+		private void seekPartitions(Collection<TopicPartition> partitions, boolean idle) {
+			Map<TopicPartition, Long> current = new HashMap<>();
+			for (TopicPartition topicPartition : partitions) {
+				current.put(topicPartition,	ListenerConsumer.this.consumer.position(topicPartition));
+			}
+			ConsumerSeekCallback callback = new ConsumerSeekCallback() {
+
+				@Override
+				public void seek(String topic, int partition, long offset) {
+					ListenerConsumer.this.consumer.seek(new TopicPartition(topic, partition), offset);
```

Comment:
It doesn't relate to `invokeListener`. 
We preform `processSeeks()` in the `ListenerConsumer.run()`. And that is the same place for the `seekPartitions(getAssignedPartitions(), true);` in case of idle. Therefore the same `Thread`.
But, yeah, I see that in case of `seeks` we will wait for the next `while (isRunning()) {` round and won't block the listener's Thread. Maybe that is a point.
Looks like the same is applicable for the logic in the `onPartitionsAssigned()`:
```
* This callback will execute in the user thread as part of the {@link Consumer#poll(long) poll(long)} call whenever partition assignment changes.
```
from `ConsumerRebalanceListener` JavaDocs.

I won't mind either way. Just want to understand why there is a difference in logic...